crates/wasi-provider/src/states/running.rs

```diff
@@ -0,0 +1,21 @@
+use kubelet::state::{State, Transition};
+use kubelet::{
+    pod::{Phase, Pod},
+    state,
+};
+
+use crate::{make_status, PodState};
+
+state!(
+    /// The Kubelet is running the Pod.
+    Running,
+    PodState,
+    {
+        // Wascc has no notion of exiting so we just sleep.
+        // I _think_ that periodically awaiting will allow the task to be interrupted.
+        loop {
```

Comment:
`wascc` has no notion of "completion" but `wasi` appears to (based on init container logic). I think it would make sense for this state to poll for completion of one or more containers and move to completed / failed accordingly.